src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,250 @@
+import { OffscreenRenderEngine } from '../engines/offscreen-render-engine';
+import { SeparatedInteractionsEngine } from '../engines/separated-interactions-engine';
+import { HitRegion, RegionTypes, Timeseries, TimeseriesChart, TooltipField } from '../types';
+import UIPlugin from './ui-plugin';
+import { mergeObjects } from '../utils';
+import { binarySearch, ChartStyle, renderChart } from './utils/chart-render';
+
+export type TimeseriesPluginStyles = {
+    height: number;
+};
+
+export const defaultTimeseriesPluginStyles: TimeseriesPluginStyles = {
+    height: 56,
+};
+
+export type TimeseriesPluginSettings = {
+    styles?: Partial<TimeseriesPluginStyles>;
+};
+
+type TimeseriesPreparedChart = TimeseriesChart & {
+    group: string;
+    style: ChartStyle;
+};
+
+type PreparedTimeseries = TimeseriesPreparedChart[];
+
+const EXTRA_POINTS_FOR_RENDER = 2;
+
+export class TimeseriesPlugin extends UIPlugin<TimeseriesPluginStyles> {
+    height = 68;
+
+    private data: PreparedTimeseries;
+    private hoveredRegion: HitRegion<{}> | null = null;
+    private summary: Record<string, { min: number; max: number; start: number; end: number }> = {};
+    private timeboxes: { start: number; end: number }[] = [];
+
+    constructor({
+        name = 'timeseriesPlugin',
+        data,
+        settings,
+    }: {
+        name?: string;
+        data: Timeseries;
+        settings?: TimeseriesPluginSettings;
+    }) {
+        super(name);
+
+        this.setSettings(settings);
+        this.setData(data);
+    }
+
+    override init(renderEngine: OffscreenRenderEngine, interactionsEngine: SeparatedInteractionsEngine) {
+        super.init(renderEngine, interactionsEngine);
+
+        this.interactionsEngine.on('change-position', this.handlePositionChange.bind(this));
+        this.interactionsEngine.on('hover', this.handleHover.bind(this));
+
+        this.interactionsEngine.on('up', this.handleMouseUp.bind(this));
+    }
+
+    handlePositionChange(position: { deltaX: number }) {
+        const startPositionX = this.renderEngine.parent.positionX;
+
+        this.interactionsEngine.setCursor('grabbing');
```

Comment:
I think we have an enum called `CursorTypes` that contains cursors used in the project